homeassistant/components/date/__init__.py

```diff
@@ -80,13 +80,13 @@ class DateEntity(Entity, cached_properties=CACHED_PROPERTIES_WITH_ATTR_):
     @final
     def device_class(self) -> None:
         """Return the device class for the entity."""
-        return None
+        return
```

Comment:
Thought => ruff was amended to keep None for properties ans cached_properties.
Should ruff also handle propcache.cached_property?

Ideally a property should always return a value (including `None`)

Reply:
> Thought => ruff was amended to keep None for properties and `functools.cached_properties`. Should ruff also handle `propcache.cached_property`?
> 

That sounds reasonable to me.  Was that change done directly in ruff?

Reply:
Yes - see https://github.com/astral-sh/ruff/pull/12563

Reply:
Since `propcache` isn't standard lib, it seems like we couldn't added it there as it would make propcache a dep

Reply:
mmm... I thought it was simply by name.
I wonder if there is a way to add "custom" property indicators via config...
I'll have a look.

Reply:
It looks like you can use this in `pyproject.toml`

```toml
[tool.ruff.lint.pydocstyle]
property-decorators = ["propcache.cached_property"]
```

Reply:
thanks. Done in the parent PR and merged here